### PR TITLE
feat: add `PolicyDefinitionNewControllerApi`

### PR DESCRIPTION
**Pull request by ndr-brt**
## What this PR changes/adds

Introduce `PolicyDefinitionNewControllerApi` and implements `createPolicyDefinition` method

## Why it does that

new protocol

## Further notes

- I kept this PR as small as possible because it gives JsonLD capabilities to the Jersey extension in its entirety. This because we can't think of having an EDC component running that's not able to interpret JsonLDs. So the `TypeManagerContextResolver` looks for the `json-ld` registered object mapper.
- checks are failing now, but in the meantime I'd like to verify the concept (will fix them later on)

## Linked Issue(s)

Part of #2768 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [x] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

**Comment by jimmarino** [comment]
@ndr-brt Can we give JSON-LD capabilities by mapped context instead of the entire Jersey instance?

File: extensions/common/http/jersey-core/src/main/java/org/eclipse/edc/web/jersey/TypeManagerContextResolver.java [hunk 1]
```diff
@@ -32,6 +32,6 @@ public TypeManagerContextResolver(TypeManager typeManager) {
 
     @Override
     public ObjectMapper getContext(Class<?> type) {
-        return typeManager.getMapper();
+        return typeManager.getMapper("json-ld");
```
Comment (ronjaquensel) [inline extensions/common/http/jersey-core/src/main/java/org/eclipse/edc/web/jersey/TypeManagerContextResolver.java:35]: the constant `TYPE_MANAGER_CONTEXT_JSON_LD` can be used here

File: extensions/control-plane/api/management-api/policy-definition-api/src/main/java/org/eclipse/edc/connector/api/management/policy/PolicyDefinitionNewApiController.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Override
+    public PolicyDefinitionResponseDto getPolicyDefinition(String id) {
+        throw new UnsupportedOperationException("not implemented yet");
+    }
+
+    @POST
+    @Override
+    public IdResponseDto createPolicyDefinition(PolicyDefinitionNewRequestDto policy) {
+        var inputDefinition = transformerRegistry.transform(policy, PolicyDefinition.class)
```
[the remark this answers is not in the document]
Comment (ndr-brt) [reply extensions/control-plane/api/management-api/policy-definition-api/src/main/java/org/eclipse/edc/connector/api/management/policy/PolicyDefinitionNewApiController.java:62]: yes, I kept the scope of this PR as little as possible to provide a generic way to use the json ld mapper on the management api controllers.

**Comment by ndr-brt** [comment]
> @ndr-brt Can we give JSON-LD capabilities by mapped context instead of the entire Jersey instance?

yes, you're right, I just wanted to provide a unique way to register an object mapper, not really having it registered for all the contexts automatically. I adapted the PR accordingly

**Pull request merged**